src/components/molecules/ProductModal.tsx

```diff
@@ -0,0 +1,36 @@
+import * as React from 'react'
+import Modal from '@mui/material/Modal'
+import { Box } from '@mui/material'
+
+const style = {
+  position: 'absolute',
+  top: '50%',
+  left: '50%',
+  transform: 'translate(-50%, -50%)',
+  width: 400,
+  bgcolor: 'background.paper',
+  border: '2px solid #000',
+  boxShadow: 24,
+  p: 4,
+}
+
+interface ModalProps {
+  open: boolean
+  handleClose: () => void
+  children: React.ReactElement
+}
+
+export const ProductModal = ({ open, handleClose, children }: ModalProps) => {
+  return (
+    <div>
+      <Modal
+        open={open}
+        onClose={handleClose}
+        aria-labelledby="modal-modal-title"
+        aria-describedby="modal-modal-description"
+      >
+        <Box sx={style}>{children}</Box>
+      </Modal>
+    </div>
```

Comment:
You create ProductModal, but its better to create reasuble shared component for Modal